### ISSUE TITLE
Filtering Error Messages

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1690]
**Is your feature request related to a problem? Please describe.**
Sometimes pyright has a false positive for a certain error. The underlying error may be reasonably caused by a 3rd party library doing import weirdness, but the errors produced end up feeling like noise when it's unlikely making good type stubs will happen for that 3rd party library. I would like a way to reduce that noise.

**Describe the solution you'd like**
A configuration option to filter errors based on the error message using a regex. My ideal solution would be something like

"ignore_error_messages": regex

edit: Also I know I'm making a number of issues. If there's a different way of reporting for small things like slack I don't mind. I'm currently working on adding types to a codebase mostly lacking them and am just reporting the various things I encounter with pyright. I'm pretty happy with pyright overall and these are just a couple small things I've noticed.

**Comment by erictraut** [comment]
Pyright already offers flexible control over individual diagnostic checks.

You can specify the severity of each check in the pyrightconfig.json file. See [here](https://github.com/microsoft/pyright/blob/master/docs/configuration.md) for details.

You can also further override the configuration on a per-file basis using a "pyright" comment at the top of the file:
```python
# pyright: reportMissingImports=false
```

**Comment by erictraut** [comment]
You can also disable diagnostics on specific lines using `# type: ignore` comments, which are defined in PEP 484.
